mne/viz/.ipynb_checkpoints/topo-checkpoint.py

```diff
@@ -0,0 +1,962 @@
+"""Functions to plot M/EEG data on topo (one axes per channel)."""
```

Comment:
I don't think you should commit this file, we don't want it in our repository. Ideally what you would do instead is modify something in `mne/viz/tests/test_topo.py` that fails on `main` but passes on this PR

Reply:
I believe this is just a copy of the original that was accidentally committed :)

Reply:
yes, we could consider adding a line `.ipynb_checkpoints/` to our [`.gitignore` file](https://github.com/mne-tools/mne-python/blob/main/.gitignore) to avoid this problem in the future.

Reply:
Done in 409e970

Reply:
FYI this part is still pending, though:

> Ideally what you would do instead is modify something in mne/viz/tests/test_topo.py that fails on main but passes on this PR

Reply:
@twang5 Let us know if you need a hand adding the test case @larsoner proposed. This stuff can sometimes be tricky for new contributors!

Reply:
@larsoner Thank you for pointing this out! @hoechenberger Thank you so much and I will look into this today - will let you know if I need help:)

Reply:
Hi, just a quick clarification: did you mean that I should add a test case to test_topo.py that tests the function _plot_evoked_topo() specifically, or should I  modify something that makes test_topo.py fail on main? 

I ran the test on this PR and on main. Both of them passed.

Reply:
> Hi, just a quick clarification: did you mean that I should add a test case to test_topo.py that tests the function _plot_evoked_topo() specifically, or should I modify something that makes test_topo.py fail on main?

Ideally you would add a new test (or add to an existing test) in `mne/viz/tests/test_topo.py` something that fails on `main` but passes on this PR (i.e., "test-driven development" / TDD). It's best if your test only uses public functions like `evoked.plot_topo(...)`, but if you can't find a way to hit the problematic code path using the public functions (this is rare!) then you can call the private `_plot_evoked_topo` to hit it. For a change like this, the best case scenario is that your new test is a single `evoked.plot_topo(...)` call where the `(...)` is filled in such that it triggers an error on `main` but passes on this PR.

You can limit what gets run locally using something like `pytest mne/viz/tests/test_topo.py -k test_plot_topo` to subselect the test(s) of interest so it's easier to test if your code does what you want. In practice, it's sometimes easier to start by writing a minimal example script that uses the `sample` dataset and hits the error, then transition to using the `testing` dataset or the builtin files that are part of the repo (I think there is a `-ave.fif` in there, too), then incorporate this in some `test_something.py` file in the repo.

Reply:
> 
> 
> Ideally you would add a new test (or add to an existing test) in `mne/viz/tests/test_topo.py` something that fails on `main` but passes on this PR (i.e., "test-driven development" / TDD). It's best if your test only uses public functions like `evoked.plot_topo(...)`, but if you can't find a way to hit the problematic code path using the public functions (this is rare!) then you can call the private `_plot_evoked_topo` to hit it. For a change like this, the best case scenario is that your new test is a single `evoked.plot_topo(...)` call where the `(...)` is filled in such that it triggers an error on `main` but passes on this PR.
> 
> You can limit what gets run locally using something like `pytest mne/viz/tests/test_topo.py -k test_plot_topo` to subselect the test(s) of interest so it's easier to test if your code does what you want. In practice, it's sometimes easier to start by writing a minimal example script that uses the `sample` dataset and hits the error, then transition to using the `testing` dataset or the builtin files that are part of the repo (I think there is a `-ave.fif` in there, too), then incorporate this in some `test_something.py` file in the repo.

Hi, thank you so much for the very detailed guidance! Unfortunately, I wasn't able to trigger the error in 'main' even when I use the private method. Sorry this is my first time doing TDD and I'd really appreciate your patience.

My idea was that the error should be triggered when the 'color' parameter is not specified, in which case the method will attempt to add '[w]' to '_get_color_list' which is invalid. I basically created an 'evoked' and tried the method without specifying 'color', but it doesn't result in an error.

Reply:
Okay it looks like this code path can never be hit by our public functions:
```
$ git grep _plot_evoked_topo
mne/viz/evoked.py:from .topo import _plot_evoked_topo
mne/viz/evoked.py:    return _plot_evoked_topo(evoked=evoked, layout=layout,
mne/viz/topo.py:def _plot_evoked_topo(evoked, layout=None, layout_scale=0.945,
```
so it's only used by `plot_evoked_topo`, and in that function we have:

https://github.com/mne-tools/mne-python/blob/89e163f60be122be2150ee93a9d3fcea5641b8b7/mne/viz/evoked.py#L866-L872

So I think the solution here is to have the color handling in exactly one place rather than two. Might as well do it in `plot_evoked_topo` since that's where the logic above lives.

Reply:
> So I think the solution here is to have the color handling in exactly one place rather than two. Might as well do it in `plot_evoked_topo` since that's where the logic above lives.

That makes sense!

Reply:
@twang5 you probably did not have time to follow up on this, so I took the liberty and pushed a few commits. I don't guarantee that this is how @larsoner expected the solution though :wink: 

See also how I changed your PR description to something more informative: This will save people one extra lookup of what the issue was about.

Finally, I edited your PR description to be more concise and to not feature the boilerplate text that is only intended for guidance during writing the description (not for submission)